select.go

```diff
@@ -13,85 +13,155 @@ import (
+	// inside the templates.
+	//
+	// For example, `{{ .Name }}` will display the name property of a struct.
 	Items interface{}
 
-	// Size is the number of items that should appear on the select before
-	// scrolling. If it is 0, defaults to 5.
+	// Size is the number of items that should appear on the select before scrolling is necessary. Defaults to 5.
 	Size int
 
-	// IsVimMode sets whether readline is using Vim mode.
+	// IsVimMode sets whether to use vim mode when using readline in the command prompt. Look at
+	// https://godoc.org/github.com/chzyer/readline#Config for more information on readline.
 	IsVimMode bool
 
 	// Templates can be used to customize the select output. If nil is passed, the
-	// default templates are used.
+	// default templates are used. See the SelectTemplates docs for more info.
 	Templates *SelectTemplates
 
-	// Keys can be used to change movement and search keys.
+	// Keys is the set of keys used in select mode to control the command line interface. See the SelectKeys docs for
+	// more info.
 	Keys *SelectKeys
 
-	// Searcher can be implemented to teach the select how to search for items.
+	// Searcher is a function that can be implemented to refine the base searching algorithm in selects.
+	//
+	// Search is a function that will receive the searched term and the item's index and should return a boolean
+	// for whether or not the terms are alike. It is unimplemented by default and search will not work unless
+	// it is implemented.
 	Searcher list.Searcher
 
-	// Starts the prompt in search mode.
+	// StartInSearchMode sets whether or not the select mdoe should start in search mode or selection mode.
+	// For search mode to work, the Search property must be implemented.
 	StartInSearchMode bool
 
 	label string
 
 	list *list.List
 }
 
-// SelectKeys defines which keys can be used for movement and search.
+// SelectKeys defines the available keys used by select mode to enable the user to move around the list
+// and trigger search mode. See the Key struct docs for more information on keys.
 type SelectKeys struct {
-	Next     Key // Next defaults to down arrow key
-	Prev     Key // Prev defaults to up arrow key
-	PageUp   Key // PageUp defaults to left arrow key
-	PageDown Key // PageDown defaults to right arrow key
-	Search   Key // Search defaults to '/' key
+	// Next is the key used to move to the next element inside the list. Defaults to down arrow key.
+	Next     Key
+
+	// Prev is the key used to move to the previous element inside the list. Defaults to up arrow key.
+	Prev     Key
+
+	// PageUp is the key used to jump back to the first element inside the list. Defaults to left arrow key.
+	PageUp   Key
+
+	// PageUp is the key used to jump forward to the last element inside the list. Defaults to right arrow key.
+	PageDown Key
+
+	// Search is the key used to trigger the search mode for the list. Default to the "/" key.
+	Search   Key
```

Comment:
IMO the previous comments were enough.

Reply:
As a newcomer, it was a bit unclear to me what some of the keys did exactly. Since I added details to pageUp/Down, I thought I'd add some everywhere.